### DISCUSSION TITLE
Process improvement request: Issue Triage

### DISCUSSION THREAD

**Discussion by tjhart** [teslamate-org/teslamate#3765] [General]
TLDR; It took a LOT of effort to get an issue looked at. Initial responses indicated the problem wasn't reviewed sufficiently. Can it be improved?

I love TeslaMate. I work to keep my installation running 24/7. It's fantastic to see the info it provides, and I want it to to be around for a long time. I'm grateful for the effort that the team members put into fixing bug requests, adding features, and maintaining documentation.

I'm a bit unhappy with the amount of effort it took to get an issue's core problem identified today. I've had to triage issues before, both public and private projects. Its frustrating. Often the reporter is new, but it's **still** the same question that's been answered in `FAQs`, `READMEs`, `TROUBLESHOOTINGs`, and `<OH-MY-GOD-CAN-YOU-READ!s>`. It's easy for the responder to pigeon-hole a problem into a set of common responses without verifying it. I've done it, to my embarrassment.

In the last several years I've found it increasingly hard to get real issues identified across several open source projects. The norm seems to be:

1. Report
2. Wait for dismissive response
3. Reply, spelling out details that seemed clear in the first report
4. Wait for second dismissive response
5. Now, point by point, refute counter arguments with data already provided in initial entry and first report.
6. Maybe get a positive response, or the conversation devolves into combative retorts, not associated with the reported issue.

Sometimes I have the energy or interest to fight to be heard, but often I'm just as stressed or busy as the project team. It can be faster to put in a hack, or even find another library/project without the bug, than to get an issue resolved. I've examined my reports to see if it's my reporting style or supporting information, but I don't think so. I will continue to reflect and see if something jumps out.

Real triage takes time, and given a public project, I assume the amount of time one has to spend on triage is tight. Especially with a group of volunteers. So maybe some sort of process or automation tools could be used to help?

What could be done to improve the efficacy of triage here, without putting an unreasonable strain on the project members?

Edited: typo

**Comment by cwanja** [comment]
Thanks for the post and candid feedback. I hear your feedback and can understand your perspective. Let's remember - this is a free, volunteer based solution. None of us are paid or get compensated and we all want to see the solution live on. But it's purely in our free time. 

I have to disagree a bit that it took 'a lot of effort' to get to the root of your issue. You posted five hours and seemed to get traction immediately. Yes, it was dejected at first. And took a few extra responses. But it took me a few times to read and reread to make sure I understood. To your bulleted points, remember what seems clear to you might not be to others. 

I am pretty confident a bot will not help. Especially given the complexity of problems we see and interpreting data, images, etc. If you have used a customer service bot recently, probably echo the same sentiment. 

If you want to see it improve, we are always looking for team members to dive in and support.

**Reply by tjhart** [reply]
It's all relative I guess.

* The initial report took 2 hours to compose, gather screen shots and data exports.
* [This reply](https://github.com/teslamate-org/teslamate/issues/3763#issuecomment-2015680841) took 20 minutes to compose
* [This reply](https://github.com/teslamate-org/teslamate/issues/3763#issuecomment-2015842495) took 90 minutes to compose
* [This reply](https://github.com/teslamate-org/teslamate/issues/3763#issuecomment-2015873192) took 20 minutes to research, gather data,  and compose
* [This reply](https://github.com/teslamate-org/teslamate/issues/3763#issuecomment-2015947923) took 40 minutes to research and compose.

4.8 hours effort. More time explaining, researching, debugging, and proposing a solution (which may or may not be the correct one) as in the original report. 

Yeah - for me, that's quite a bit of work to explain something. I don't mean to imply that it's an unreasonable amount for everyone, or for the average person. But after that, I want a nap.

**Reply by tjhart** [reply]
@cwanja I wasn't thinking of any kind of bot - more like scripts and tools to set up an environment, so the team member has a `teslamate` instance up and running with provided data, so they can see the graph and underlying database tables without having to click more than one button or type more than one command.
